src/components/SingleArticle/SingleArticle.jsx

```diff
@@ -0,0 +1,33 @@
+import { useEffect, useState } from "react";
+import { useLocation, useParams } from "react-router-dom"
+import { fetchArticleById } from "../../api/api";
+import ArticleSection from "../ArticleSection/ArticleSection";
+
+function SingleArticle() {
+  const [isLoading, setIsLoading] = useState(false)
+  const [singleArticle, setSingleArticle] = useState({})
+  const { article } = useParams();
+  const articleId = article.split('-').slice(-1)
```

Comment:
This is an odd way to get the articleId - read the notes on useParams and react router to refresh your memory.